plugins/gatsby-remark-args-linker/index.js

```diff
@@ -0,0 +1,74 @@
+const visit = require('unist-util-visit')
+
+const argsRegex = new RegExp(/\-{1,2}[a-zA-Z-]*/, 'ig')
+
+function patch(context, key, value) {
+  if (!context[key]) {
+    context[key] = value
+  }
+
+  return context[key]
+}
+const svgIcon = `<svg aria-hidden="true" focusable="false" height="16" version="1.1" viewBox="0 0 16 16" width="16"><path fill-rule="evenodd" d="M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z"></path></svg>`
+
+module.exports = (
+  { markdownAST },
+  { icon = svgIcon, className = 'anchor', isIconAfterHeader = false }
+) => {
+  visit(
+    markdownAST,
+    node =>
+      node.type === 'listItem' &&
+      node.children.some(
+        item =>
+          item.type === 'paragraph' &&
+          item.children[0]?.type === 'inlineCode' &&
+          String(item.children[0]?.value).startsWith('-')
+      ),
+    listItemNode => {
+      const paragraphNode = listItemNode.children.find(
+        item => item.type === 'paragraph'
+      )
+      const inlineCodeNode = paragraphNode.children.find(
+        item => item.type === 'inlineCode'
+      )
+      const value = inlineCodeNode.value
+      if (String(value).startsWith('-')) {
+        const id = value.match(argsRegex)[0]
+        const data = patch(listItemNode, `data`, {})
+
+        patch(data, `id`, id)
+        patch(data, `htmlAttributes`, {})
+        patch(data, `hProperties`, {})
+        patch(data.htmlAttributes, `id`, id)
+        patch(data.hProperties, `id`, id)
+        if (icon) {
```

Comment:
[Q] why do we need this `if`?

Reply:
icon is taken as parameter for plugin so if we don't want to set it we can pass false or empty value on config.

Reply:
let's simplify this, wdyt? we don't need to pass an icon to the plugin as far as I understand. We always use the same icon

Reply:
This one was also kind of discussed in another conversation as well. We use `gatsby-autolink-headers` as well. So, for consistency, we defined the link-icon separately and passed that icon as parameters on both.